tabakins2html.py

```diff
@@ -0,0 +1,125 @@
+from railroad import Diagram, Choice, Comment, NonTerminal, Sequence, Terminal, Skip, Optional, ZeroOrMore, OneOrMore
+import sys
+import pymustache
+import io
+
+ALLOWED_NAMES = {
+    "Comment": Comment,
+    "Diagram": Diagram, 
+    "Choice": Choice, 
+    "Comment": Comment, 
+    "NonTerminal": NonTerminal, 
+    "Sequence": Sequence, 
+    "Terminal": Terminal,
+    "Skip": Skip,
+    "Optional": Optional,
+    "ZeroOrMore": ZeroOrMore,
+    "OneOrMore": OneOrMore,
+    "undefined": 'undefined'
+}
+
+def eval_expression(input_string, allowed_names = {}):
+    code = compile(input_string, "<string>", "eval")
+    for name in code.co_names:
+        if name not in allowed_names:
+            raise NameError(f"Use of {name} not allowed")
+    return eval(code, {"__builtins__": {}}, allowed_names)
+
+def pars_tadatkins_file():
```

Comment:
it's tabatkins (same issue persistent throughout)

Reply:
plus pars -> parse

Reply:
Fixed - thanks - I had a bit of dyslexia there!